### PR TITLE
fix(SearchBar): onCancel called twice issue on platform android

### PR DESCRIPTION
**Pull request by kedar09**
## Motivation

Fixes # (issue)
https://github.com/react-native-elements/react-native-elements/issues/3247
## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [x] Checked with `example` app

## Checklist

- [x] My code follows the style guidelines of this project
- [x] My changes generate no new warnings

### CODE REVIEW

File: packages/base/src/SearchBar/SearchBar-android.tsx [hunk 1]
```diff
@@ -103,19 +92,6 @@ export class SearchBarAndroid extends Component<
...
     };
-    this.keyboardListener = Keyboard.addListener(
-      'keyboardDidHide',
-      this._keyboardDidHide
-    );
-  }
-  _keyboardDidHide = () => {
-    this.cancel();
-  };
-
```
Comment (arpitBhalla) [inline packages/base/src/SearchBar/SearchBar-android.tsx]: Hey, how removing this part of code will fix the issue?

Comment (kedar09) [reply packages/base/src/SearchBar/SearchBar-android.tsx]: https://github.com/react-native-elements/react-native-elements/issues/3247#issue-1055081036 you can check this video,
when clicking on the back icon button, the cancel method is called two times 
1. on onPress
    https://github.com/react-native-elements/react-native-elements/blob/ad93c92924d9ccbaf8ab29da417cb2ea08589463/packages/base/src/SearchBar/SearchBar-android.tsx#L168-L175
2. on keyboardHide
    https://github.com/react-native-elements/react-native-elements/blob/ad93c92924d9ccbaf8ab29da417cb2ea08589463/packages/base/src/SearchBar/SearchBar-android.tsx#L111-L113
onCancel will work the same as the Cancel Button of platform ios

Comment (arpitBhalla) [reply packages/base/src/SearchBar/SearchBar-android.tsx]: But then onCancel will not get triggered if we hide keyboard manually, maybe we can add `Keyboard.dismiss()` on LeftIcon press?

Comment (kedar09) [reply packages/base/src/SearchBar/SearchBar-android.tsx]: I have added Keyboard.dismiss(), it worked properly.

File: packages/base/src/SearchBar/SearchBar-android.tsx [hunk 2]
```diff
@@ -169,7 +169,7 @@ export class SearchBarAndroid extends Component<
             hasFocus
               ? renderNode(Icon, cancelIcon, {
                   ...defaultCancelIcon(theme as Theme),
-                  onPress: this.cancel,
+                  onPress: () => Keyboard.dismiss(),
```
Comment (arpitBhalla) [inline packages/base/src/SearchBar/SearchBar-android.tsx]: This can be `onPress:Keyboard.dismiss`

**Comment by arpitBhalla** [comment]
Can you test it on web?

**Pull request closed without merging**